docs/image-example.rst

```diff
@@ -0,0 +1,253 @@
+with the following keys:
+
+``name``
+   The name designated to the datum.  This is used both for connecting tasks
+   together in a workflow and, in the case of python tasks, the name of the
+   variable injected into/extracted from the tasks scope.
+
+``type``
+   The general data type expected by the task.  See :ref:`types-and-formats`
+   for a list of types provided by Romanesco's core library as well as
+   :ref:`romanesco-plugins` for additional data types provided by optional
+   plugins.
+
+``format``
+   The specific representation or encoding of the data type.  Romanesco will
+   automatically convert between different data formats provided that they
+   are of the same base type.
+
+.. testcode::
+
+   show_image = {
+       'inputs': [{'name': 'the_image', 'type': 'image', 'format': 'pil'}],
+       'outputs': [],
+       'script': 'the_image.show()'
+   }
+
+In order to run the task, we will need to provide an :ref:`input binding <input-spec>`
+that tells Romanesco where it can get the data to be injected into the port.  Several
+I/O modes are supported; in this case, we provide a public URL to an image that
+Romanesco will download and open using Pillow.  Notice that Romanesco downloads and
+reads the file as part of the automatic data format conversion.
+
+.. testcode::
+
+   lenna = {
+       'type': 'image',
+       'format': 'png',
+       'url': 'https://upload.wikimedia.org/wikipedia/en/2/24/Lenna.png'
+   }
+
+.. testcode::
+   :hide:
+
+   # here we replace the show image task to a no-op so we can test without
+   # a display
+   show_image = {
+       'inputs': [{'name': 'the_image', 'type': 'image', 'format': 'pil'}],
+       'outputs': [{'name': 'the_image', 'type': 'image', 'format': 'pil'}],
+       'script': 'pass'
+   }
+
+Finally to run this task, we only need to provide the task object and the input binding to
+:func:`romanesco.run`.  The object returned by this function contains data extracted
+and converted through the task's output ports.
+
+.. testcode::
+
+   output = romanesco.run(show_image, {'the_image': lenna})
+
+.. doctest::
+   :hide:
+
+   >>> isinstance(output['the_image']['data'], Image)
+   True
+
+
+Perform an image blur inside a workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that we know how to generate a simple task using Romanesco, we
+want to learn how to connect multiple tasks together in a workflow.
+Romanesco's pythonic API allows us to do this easily.  Lets create
+a new task that performs a blur operation on an image.  This might
+look like the following:
+
+.. testcode::
+
+   blur_image = {
+      'inputs': [
+         {'name': 'blur_input', 'type': 'image', 'format': 'pil'},
+         {'name': 'blur_radius', 'type': 'number', 'format': 'number'}
+      ],
+      'outputs': [{'name': 'blur_output', 'type': 'image', 'format': 'pil'}],
+      'script': '''
+   from PIL import ImageFilter
+   blur_output = blur_input.filter(ImageFilter.GaussianBlur(blur_radius))
+   '''
+   }
```

Comment:
@jbeezley @jeffbaumes You may be interested in looking at the following task https://github.com/Kitware/minerva/blob/mean-contour-workflow/analyses/NEX/mean_contour_workflow/classes.py#L157-L194  which is apart of a workflow that looks like this:

```python
    wf = Workflow()

    wf.add_task(NetCDFMean(), "m")
    wf.add_task(NetCDFToContourJson(), "c")

    wf.connect_tasks("m", "c", {"output": "data"})

    c = GirderClient('localhost', 8081)
    c.authenticate("kotfic", "letmein")

    output = romanesco.run(
        wf,
        inputs={
             "file_path": {
                 'mode': 'girder',
                 'host': 'localhost',
                 'scheme': 'http',
                 'port': 8081,
                 'api_root': '/api/v1',
                 'resource_type': 'file',
                 'id': "55df4fbaf4b1496693a3c464",
                 'type': 'string',
                 'format': 'text',
                 "name": "test.nc",
                 'token': c.token
             },
            "m.variable": {
                "format": "text",
                "data": "pr"},
            "c.variable": {
                "format": "text",
                "data": "pr"}},
        outputs={
            'contour': {
                'mode': 'girder',
                'name': 'test.json',
                'host': 'localhost',
                'scheme': 'http',
                'port': 8081,
                'api_root': '/api/v1',
                'parent_type': 'folder',
                'parent_id': "55b13466f4b149110f800aa1",
                'format': 'json',
                'type': 'string',
                'token': c.token
            }})

```

This relies on classes defined here: https://github.com/Kitware/minerva/blob/mean-contour-workflow/analyses/NEX/base/__init__.py  which encapsulate the task script handling.  This allows classes inheriting from ```MinervaTask``` to define a ```run``` function with arity equal to the number of input ports. This is still very much a work in progress.